composer/lib/dependabot/composer/update_checker/latest_version_finder.rb

```diff
@@ -143,7 +143,23 @@ def parse_registry_response(response, url)
           return [] if listing.fetch("packages", []) == []
           return [] unless listing.dig("packages", dependency.name.downcase)
 
-          listing.dig("packages", dependency.name.downcase).keys
+          version_listings = listing.dig("packages", dependency.name.downcase)
+          # Packagist's original v1 Metadata API was of the form:
+          # * "packages": {<package name>: {<version number>: {hash of metadata for a particular release version}}}
+          #
+          # The updated v2 Metadata API is of the form:
+          # * "packages": {<package name>: [{hash of metadata for a particular release version}]}
+          #
+          # More details:
+          # * https://github.com/dependabot/dependabot-core/issues/3010
+          # * https://github.com/composer/composer/blob/master/UPGRADE-2.0.md#for-composer-repository-implementors
+          if version_listings.is_a?(Array) # default to v2 format
+            version_listings.map { |i| i.fetch("version") }
```

Comment:
I think you could simply do this for all registries as even those with a hash keyed by versions must contain the version in the metadata hash, so this should be compatible with everything (tho a tiny bit slower than simply returning `.keys` I am sure)

Reply:
Note that composer itself completely ignores the keys in the version listing. So it would be safer to always read it from the metadata hash as repositories other than packagist.org could be returning different values.

Reply:
👍 switched to coercing v1 results to an array of metadata hashes to match v2 format.